python/lsst/daf/butler/core/formatter.py

```diff
@@ -60,10 +64,11 @@ class Formatter(metaclass=ABCMeta):
     are supported (`frozenset`).
     """
 
-    def __init__(self, fileDescriptor: FileDescriptor):
+    def __init__(self, fileDescriptor: FileDescriptor, dataId: "DataCoordinate" = None):
```

Comment:
If you use `from __future__ import annotations`, you can write `DataCoordinate` without the quotes.